src/Simplex/Chat.hs

```diff
@@ -5315,6 +5365,29 @@ processAgentMessageConn vr user@User {userId} corrId agentConnId agentMessage =
     checkHostRole GroupMember {memberRole, localDisplayName} memRole =
       when (memberRole < GRAdmin || memberRole < memRole) $ throwChatError (CEGroupContactRole localDisplayName)
 
+    xGrpMemBlock :: GroupInfo -> GroupMember -> MemberId -> Bool -> RcvMessage -> UTCTime -> m ()
+    xGrpMemBlock gInfo@GroupInfo {groupId, membership} m@GroupMember {memberRole = senderRole} memId blocked msg brokerTs
+      | membershipMemId == memId =
+          -- member shouldn't receive this message about themselves
+          messageError "x.grp.mem.block: admin blocks you"
+      | otherwise =
+          withStore' (\db -> runExceptT $ getGroupMemberByMemberId db user gInfo memId) >>= \case
```

Comment:
as you fail on absent member anyway, maybe no need to handle it here, and can just use withStore without runExceptT?

Reply:
also, can there be a race between member being informed about all members and some members being blocked? In which case do we need to store these blockings? or, at least, include the information about member being blocked in introductions? also the member can be blocked before new member joins - I don't see it handled.

Reply:
So you still end up with the need to communicate blocking as part of profile?